### ISSUE TITLE
aws_wafv2_web_acl - Error: Provider produced inconsistent final plan

### ISSUE THREAD

**Issue by manfro3** [hashicorp/terraform-provider-aws#23992]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Related:
- #23390 (42)
- #23423 (16)
- #23936 (16)
- 👉 #23992 (117)
- #24386 (13)
- #27175 (27)
- #27273 (42)
- #27479 (20)
- #28191 (34)
- #28672 (23)
- #29012 (9)
- #29304 (43)
- #30858 (5)

### Terraform CLI and Terraform AWS Provider Version

Terraform version 1.1.7
provider registry.terraform.io/hashicorp/aws v4.8.0

### Affected Resource(s)

* aws_wafv2_web_acl

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_wafv2_ip_set" "ip_whitelist" {
  name = "IP-whitelist"
  provider  = aws.virginia
  scope = "CLOUDFRONT"
  ip_address_version = "IPV4"
  addresses = var.stage == "prod" || var.stage == "cons" ? ["xxx.xxx.xxx.xxx/32"] : ["0.0.0.0/1", "128.0.0.0/1"]
}

resource "aws_wafv2_web_acl" "whitelist_waf_acls" {
  name = "Whitelist-WAF-ACLs"
  provider = aws.virginia
  scope = "CLOUDFRONT"

  default_action {
    block {}
  }

  rule {
    name = "Whitelist-WAF-ACLs-rule"
    priority = 0

    action {
      allow {}
    }

    statement {
      ip_set_reference_statement { 
        arn = aws_wafv2_ip_set.ip_whitelist.arn
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name = "Whitelist-WAF-ACLs-rule"
      sampled_requests_enabled = true
    }
  }

  rule {
    name     = "AWSManagedRulesBotControlRuleSet"
    priority = 1

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesBotControlRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesAdminProtectionRuleSet"
    priority = 2

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesAdminProtectionRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesAmazonIpReputationList"
    priority = 3

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesAmazonIpReputationList"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesAnonymousIpList"
    priority = 4

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesAnonymousIpList"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesCommonRuleSet"
    priority = 5

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesCommonRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesKnownBadInputsRuleSet"
    priority = 6

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesKnownBadInputsRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesSQLiRuleSet"
    priority = 7

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesSQLiRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  visibility_config {
    cloudwatch_metrics_enabled = true
    metric_name = "Whitelist-WAF-ACLs-metric"
    sampled_requests_enabled = true
  }

  tags = {
    Name = "whitelist_waf_acls"
  }

  depends_on = [aws_wafv2_ip_set.ip_whitelist]
}
```

### Expected Behavior

I am trying to make a update to the config, this particular change is for removing tags

```hcl
Terraform will perform the following actions:

  # aws_wafv2_web_acl.whitelist_waf_acls will be updated in-place
  ~ resource "aws_wafv2_web_acl" "whitelist_waf_acls" {

        id         = "xxxxxx-xxxxxx-xxxxx-xxxxxx"

        name       = "Whitelist-WAF-ACLs"

      ~ tags       = {
          - "Application" = "xxxx" -> null
          - "Cost Center" = "xxxx" -> null
          - "Environment" = "dev" -> null
          - "Owner"       = "xxxxx@xxx.xx" -> null
          - "Project"     = "xxxxx" -> null
          - "TeamName"    = "xxxxx" -> null
          - "Entity" = "xxx" -> null
            # (1 unchanged element hidden)
        }
      ~ tags_all   = {
          - "Application" = "xxxx" -> null
          - "Cost Center" = "xxx" -> null
          - "Environment" = "dev" -> null
          - "Owner"       = "xxxx@xx.xx" -> null
          - "Project"     = "xxxx" -> null
          - "TeamName"    = "xxxx" -> null
          - "Entity" = "xxx" -> null
            # (1 unchanged element hidden)
        }
```

### Actual Behavior

It does not deploy, it throws the error;

```hcl
When expanding the plan for aws_wafv2_web_acl.whitelist_waf_acls to include new values learned so far during apply, provider "registry.terraform.io/hashicorp/aws" produced an invalid new value for .rule: planned set element

cty.ObjectVal(map[string]cty.Value{"action":cty.ListValEmpty(cty.Object(map[string]cty.Type{"allow":cty.List(cty.Object(map[string]cty.Type{"custom_request_handling":cty.List(cty.Object(map[string]cty.Type{"insert_header":cty.Set(cty.Object(map[string]cty.Type{"name":cty.String, "value":cty.String}))}))})), "block":cty.List(cty.Object(map[string]cty.Type{"custom_response":cty.List(cty.Object(map[string]cty.Type{"custom_response_body_key":cty.String, "response_code":cty.Number,
.
.
.
"text_transformation":cty.Set(cty.Object(map[string]cty.Type{"priority":cty.Number, "type":cty.String}))}))})}), "visibility_config":cty.ListVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{"cloudwatch_metrics_enabled":cty.True, "metric_name":cty.StringVal("Whitelist-WAF-ACLs-menaged-rule"), "sampled_requests_enabled":cty.True})})}) does not correlate with any element in actual.

 This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

### Steps to Reproduce

1. `terraform apply`

**Comment by b-dean** [comment]
I also was removing tags, I had moved some to the default tags for the provider

```hcl
provider "aws" {
  region = "us-east-1"
  default_tags {
    tags = {
      stack   = terraform.workspace
      purpose = "example"    
    }
  }
}
```

I found that if I do _only_ that, it fails with the error you mentioned. But if I also make some other arbitrary change to the web acl rules, such as changing a rule priority slightly, then it does not fail.

**Comment by gurpalw** [comment]
this is. happening to me on the latest 4.23.0 aws provider version.

**Comment by chroju** [comment]
Same issue here. I have tried to update `default_tags`, but Terraform has panicked with the same error message `When expanding the plan for aws_wafv2_web_acl.XXX to include new values...` .

This may be related to #23423 and #24386 . I work around the error by using `ignore_changes` , referring to the following link.

ref: https://github.com/hashicorp/terraform-provider-aws/issues/24386#issuecomment-1109340765

**Comment by Rumbles** [comment]
Every time we apply, we update a timestamp tag which is in the provider defaults tags, if there are changes to the WAF rules, we can apply without issues, but if the only change is a tag being updated we get this panic. I noticed this after upgrading from this provider from v3 to v4, we haven't changed the use of `aws_wafv2_web_acl` but now we get a HUGE error message (30+ pages of unhelpful error text) when ever we try to apply our resource without any changes (other than a minor tag update)

**Comment by aholthagerty** [comment]
Bumping this to keep the bots at bay.

**Comment by unclok** [comment]
As a workaround, I've just changed priority of rules, so let terraform also modify rules with tags.
I hope this problem will be fixed soon.

**Comment by Eronarn** [comment]
Also impacted by this with `v4.29.0`, in this case "only" changing the default action from "block" to "allow":

```
Terraform will perform the following actions:

  # module.cloudfront_wafv2.aws_wafv2_web_acl.main will be updated in-place
  ~ resource "aws_wafv2_web_acl" "main" {
        id          = "3a01400a-0c69-443a-a1b2-629a3156587c"
        name        = "frontend-production-cloudfront-app-waf"
        tags        = {}
        # (6 unchanged attributes hidden)

      ~ default_action {
          + allow {
            }

          - block {
            }
        }


        # (10 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

**Comment by mmaetzler** [comment]
FYI: This issues with WAFv2 has been reported multiple times: 

- #23390
- #23423
- #23936
- #23992
- #24386
- #27175
- #27273
- #27479

**Comment by amitsamal94** [comment]
when can this be fixed?

**Comment by adrian-is-on-github** [comment]
Came here to post that changing one of the waf rule priorities manually, then running plan/apply succeeded for me as a workaround (AWS Provider v2.52.0 ).

**Comment by Rumbles** [comment]
> Came here to post that changing one of the waf rule priorities manually, then running plan/apply succeeded for me as a workaround (AWS Provider v2.52.0 ).

Yes the nature of the issue is if there are NO changes and you apply you get this error, if there are any changes that are required, there is no issue

But making manual changes so that an automated process does not error isn't ideal

**Comment by adrian-is-on-github** [comment]
Agreed - not ideal. In my case we have 19 similar environments with only two encountering this error.

**Comment by bemillenium** [comment]
I also face this issue when I try to add more rule by using the dynamic block
```Error: Provider produced inconsistent final plan
│
│ When expanding the plan for module.waf_alb.aws_wafv2_web_acl.this to include new values learned so far during apply, provider
│ "registry.terraform.io/hashicorp/aws" produced an invalid new value for .rule: planned set element
│ cty.ObjectVal(map[string]cty.Value{"action":cty.ListValEmpty(cty.Object(map[string]cty.Type{"allow":cty.List(cty.Object(map[string]cty.Type{"custom_request_handling":cty.List(cty.Object(map[string]cty.Type{"insert_header":cty.Set(cty.Object(map[string]cty.Type{"name":cty.String,
│ "value":cty.String}))}))})),
│ "block":cty.List(cty.Object(map[string]cty.Type{"custom_response":cty.List(cty.Object(map[string]cty.Type{"custom_response_body_key":cty.String,
│ "response_code":cty.Number, "response_header":cty.Set(cty.Object(map[string]cty.Type{"name":cty.String, "value":cty.String}))}))})),
│ "captcha":cty.List(cty.Object(map[string]cty.Type{"custom_request_handling":cty.List(cty.Object(map[string]cty.Type{"insert_header":cty.Set(cty.Object(map[string]cty.Type{"name":cty.String,
│ "value":cty.String}))}))})),
│ "challenge":cty.List(cty.Object(map[string]cty.Type{"custom_request_handling":cty.List(cty.Object(map[string]cty.Type{"insert_header":cty.Set(cty.Object(map[string]cty.Type{"name":cty.String,
│ "value":cty.String}))}))})),
│ "count":cty.List(cty.Object(map[string]cty.Type{"custom_request_handling":cty.List(cty.Object(map[string]cty.Type{"insert_header":cty.Set(cty.Object(map[string]cty.Type{"name":cty.String,```

**Comment by hypsrizzari** [comment]
> To add I am hitting this same error on almost every change. Using provider "hashicorp/aws v4.50.0"
> 
> My workaround is to change the priority of all of the rules and then the change will apply which is awkward as the number of rules increases.
> 
> Anecdotally i only started to see this issue after migrating from exclude to rule_action_override blocks for the managed rule group statement.

@stevenwilliamson 

I had the same issue.  See this comment which worked for me. Hope it helps.  Kudos to the author.

https://github.com/hashicorp/terraform-provider-aws/issues/28191#issuecomment-1398853194

**Comment by apagliara** [comment]
Upgrading terraform to 1.4.x fixed this for me

**Comment by Rumbles** [comment]
I upgraded to 1.4 last week, but my errors persist

**Comment by Rumbles** [comment]
Looks like I was wrong, the errors in the pipeline last week must have been from something else. I was able to apply the same resource with no changes without a stack trace being generated. But it looks like it's because it is applying changes to the waf each time, regardless of whether there are real changes. But regardless, the stack traces have stopped for now.

**Comment by YakDriver** [comment]
**NOTE:** I cannot reproduce this error after trying various configurations. _**Let us know**_ if this is still a problem! If we don't hear and can't reproduce, we plan to close this on or around **July 20, 2023**.

I had no problems (no errors or inconsistent final plans) with these configurations. I applied Config 1 (tags + default tags), then Config 2 (removes tags, removes default tags), then Config 1 again. (Region must be `us-east-1`.)

Using:
- Terraform v1.5.2
- AWS Provider v5.7.0

### Config 1

```terraform
provider "aws" {
  default_tags {
    tags = {
      fine_night = "tonight"
      fine_day   = "tomorrow"
    }
  }
}

resource "aws_wafv2_ip_set" "test" {
  name               = "wafv23992test"
  scope              = "CLOUDFRONT"
  ip_address_version = "IPV4"
  addresses          = ["1.2.3.4/32", "5.6.7.8/32"]
}

resource "aws_wafv2_web_acl" "test" {
  name  = "wafv23992test"
  scope = "CLOUDFRONT"

  default_action {
    block {}
  }

  rule {
    name     = "Whitelist-WAF-ACLs-rule"
    priority = 0

    action {
      allow {}
    }

    statement {
      ip_set_reference_statement { 
        arn = aws_wafv2_ip_set.test.arn
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesBotControlRuleSet"
    priority = 1

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesBotControlRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesAdminProtectionRuleSet"
    priority = 2

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesAdminProtectionRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesAmazonIpReputationList"
    priority = 3

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesAmazonIpReputationList"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesAnonymousIpList"
    priority = 4

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesAnonymousIpList"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesCommonRuleSet"
    priority = 5

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesCommonRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesKnownBadInputsRuleSet"
    priority = 6

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesKnownBadInputsRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesSQLiRuleSet"
    priority = 7

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesSQLiRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  visibility_config {
    cloudwatch_metrics_enabled = true
    metric_name = "Whitelist-WAF-ACLs-metric"
    sampled_requests_enabled = true
  }

  tags = {
    Name        = "wafv23992test"
    Application = "wafv23992test"
    Cost        = "wafv23992test"
    Environment = "wafv23992test"
    Owner       = "wafv23992test"
    Project     = "wafv23992test"
    TeamName    = "wafv23992test"
    Entity      = "wafv23992test"
  }

  depends_on = [aws_wafv2_ip_set.test]
}
```

### Config 1

```terraform
resource "aws_wafv2_ip_set" "test" {
  name               = "wafv23992test"
  scope              = "CLOUDFRONT"
  ip_address_version = "IPV4"
  addresses          = ["1.2.3.4/32", "5.6.7.8/32"]
}

resource "aws_wafv2_web_acl" "test" {
  name  = "wafv23992test"
  scope = "CLOUDFRONT"

  default_action {
    block {}
  }

  rule {
    name     = "Whitelist-WAF-ACLs-rule"
    priority = 0

    action {
      allow {}
    }

    statement {
      ip_set_reference_statement { 
        arn = aws_wafv2_ip_set.test.arn
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesBotControlRuleSet"
    priority = 1

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesBotControlRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesAdminProtectionRuleSet"
    priority = 2

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesAdminProtectionRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesAmazonIpReputationList"
    priority = 3

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesAmazonIpReputationList"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesAnonymousIpList"
    priority = 4

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesAnonymousIpList"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesCommonRuleSet"
    priority = 5

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesCommonRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesKnownBadInputsRuleSet"
    priority = 6

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesKnownBadInputsRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  rule {
    name     = "AWSManagedRulesSQLiRuleSet"
    priority = 7

    override_action {
      count {}
    }

    statement {
      managed_rule_group_statement {
        name        = "AWSManagedRulesSQLiRuleSet"
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      metric_name                = "Whitelist-WAF-ACLs-menaged-rule"
      sampled_requests_enabled   = true
    }
  }

  visibility_config {
    cloudwatch_metrics_enabled = true
    metric_name                = "Whitelist-WAF-ACLs-metric"
    sampled_requests_enabled   = true
  }

  depends_on = [aws_wafv2_ip_set.test]
}
```

**Comment by 88lexd** [comment]
Using my sample code and how to reproduce is still occurring, see issue: https://github.com/hashicorp/terraform-provider-aws/issues/28672

**Comment by YakDriver** [comment]
@88lexd Thank you for feedback. See my response [here](https://github.com/hashicorp/terraform-provider-aws/issues/28672#issuecomment-1638620712).

Based on a combination of changes in Terraform core and the AWS provider, we believe these issues have been resolved. But, the fix requires that you upgrade to Terraform v1.4+ and the AWS provider v5.7.0+.

**Comment by justinretzolk** [comment]
Hi all :wave: As was mentioned above, this issue appears to be fixed when using a minimum Terraform version of 1.4.2 and a minimum AWS Provider version of 4.67.0 (preferably Terraform 1.5.3 or later and AWS Provider 5.8.0 or later). If you experience additional unexpected behaviors with versions that meet these parameters, please open a new issue so that we can investigate further.
